AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthModeStrategy.swift

```diff
@@ -188,19 +195,34 @@ public class AWSMultiAuthModeStrategy: AuthModeStrategy {
     /// - Returns: an iterator for the applicable auth rules
     public func authTypesFor(schema: ModelSchema,
                              operation: ModelOperation) async -> AWSAuthorizationTypeIterator {
-        var applicableAuthRules = schema.authRules
-            .filter(modelOperation: operation)
-            .sorted(by: AWSMultiAuthModeStrategy.comparator)
+        return await authTypesFor(schema: schema, operations: [operation])
+    }
+    
+    /// Returns the union of authorization types for the provided schema for the given list of operations
+    /// - Parameters:
+    ///   - schema: model schema
+    ///   - operations: model operations
+    /// - Returns: an iterator for the applicable auth rules
+    public func authTypesFor(schema: ModelSchema,
+                             operations: [ModelOperation])  async -> AWSAuthorizationTypeIterator {
+        var applicableAuthRules = Set<AuthRule>()
+        for operation in operations {
+            let rules = schema.authRules.filter(modelOperation: operation)
+            applicableAuthRules = applicableAuthRules.union(Set(rules))
+        }
```

Comment:
The overhead of using `Set` here for such a small amount of possible cases seems to negate any gain. Especially since it requires adding `Hashable` conformance to `AuthRule`.

Here's a quick example of what I mean:
```swift
// set_rules.swift
func test(schema: ModelSchema, operations: [ModelOperation]) -> [AuthRule] { 
    var applicableAuthRules = Set<AuthRule>()
    for operation in operations {
        let rules = schema.authRules.filter(modelOperation: operation)
        applicableAuthRules = applicableAuthRules.union(Set(rules))
    }

    let sortedRules = rules.sorted(by: AWSMultiAuthModeStrategy.comparator)
    return sortedRules
}

for i in 1...1_000_000 {
    _ = rules(schema: schema, operations: operations(count: i))
}
```
vs.
```swift
// array_rules.swift
func test(schema: ModelSchema, operations: [ModelOperation]) -> [AuthRule] { 
    let rules = operations
        .flatMap { schema.authRules.filter(modelOperation: $0) }
        .reduce(into: [AuthRule](), { array, rule in
            if !array.contains(rule) {
                array.append(rule)
            }
        })
        .sorted(by: AWSMultiAuthModeStrategy.comparator)

    return rules
}

for i in 1...1_000_000 {
    _ = rules(schema: schema, operations: operations(count: i))
}
```

```bash
> swiftc -o set_rules -O set_rules.swift
> swiftc -o array_rules -O array_rules.swift

> time ./set_rules
./set_rules  1.28s user 0.00s system 99% cpu 1.298 total

> time ./array_rules
./array_rules  0.77s user 0.01s system 97% cpu 0.792 total
```

I get that this may be a bit of a premature optimization, so I'm fine if you want to leave it as.